unitTests/test_types.cpp

```diff
@@ -47,15 +47,48 @@ TEST(ExivTime, doesNotGetTimeWithBadFormedString)
 TEST(DataBuf, pointsToNullByDefault)
 {
     DataBuf instance;
-    ASSERT_EQ(nullptr, instance.pData_);
-    ASSERT_EQ(0,    instance.size_);
+    ASSERT_EQ(nullptr, instance.c_data());
+    ASSERT_EQ(0,    instance.size());
 }
 
 TEST(DataBuf, allocatesDataWithNonEmptyConstructor)
 {
     DataBuf instance (5);
-    ASSERT_NE(static_cast<byte *>(nullptr), instance.pData_);  /// \todo use nullptr once we move to c++11
-    ASSERT_EQ(5,    instance.size_);
+    ASSERT_NE(static_cast<byte *>(nullptr), instance.c_data());  /// \todo use nullptr once we move to c++11
+    ASSERT_EQ(5,    instance.size());
+}
+
+// Test methods like DataBuf::read_uint32 and DataBuf::write_uint32.
+TEST(DataBuf, read_write_endianess)
```

Comment:
In this test you are checking the "happy" cases. It would also be good to create additional tests to check what happens on the "non-happy" ones:

- Offset out of limits
- passing null pointers
- etc